### PR TITLE
fix qemu firmware path locations

### LINKED ISSUE #14176: Mac M1 , Qemu2 Driver Not found
### What Happened?

I ran minikube using qemu2 driver but I am getting a PROVIDER_QEMU2_NOT_FOUND error

``` shell
minikube start --driver=qemu2 --memory=26.5g --disk-size=270g --cpus=8 \
 --addons="[dashboard,default-storageclass,ingress,ingress-dns,metrics-server,registry,storage-provisioner]" \
 --kubernetes-version=1.23.6 --wait=all --dns-proxy=false --host-dns-resolver=true \
 --insecure-registry="10.0.0.0/16" --insecure-registry="192.168.49.0/24"
😄  minikube v1.26.0-beta.0 on Darwin 12.0.1 (arm64)
✨  Using the qemu2 (experimental) driver based on user configuration

🤷  Exiting due to PROVIDER_QEMU2_NOT_FOUND: The 'qemu2' provider was not found: exec: "qemu-system-aarch64": executable file not found in $PATH
💡  Suggestion: Install qemu-system
📘  Documentation: https://minikube.sigs.k8s.io/docs/reference/drivers/qemu2/
```
So I installed qemu using homebrew and now I am getting the following error

```shell
minikube start --driver=qemu2 --memory=26.5g --disk-size=270g --cpus=8 \
 --addons="[dashboard,default-storageclass,ingress,ingress-dns,metrics-server,registry,storage-provisioner]" \
 --kubernetes-version=1.23.6 --wait=all --dns-proxy=false --host-dns-resolver=true \
 --insecure-registry="10.0.0.0/16" --insecure-registry="192.168.49.0/24"
😄  minikube v1.26.0-beta.0 on Darwin 12.0.1 (arm64)
✨  Using the qemu2 (experimental) driver based on user configuration

🤷  Exiting due to PROVIDER_QEMU2_NOT_FOUND: The 'qemu2' provider was not found: stat /opt/homebrew/Cellar/qemu/6.2.0_1/share/qemu/edk2-aarch64-code.fd: no such file or directory
💡  Suggestion: Install uefi firmware
📘  Documentation: https://minikube.sigs.k8s.io/docs/reference/drivers/qemu2/

```

### Attach the log file

```shell
Binary: Built with gc go1.18.1 for darwin/arm64
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
I0516 14:24:13.513114   28241 out.go:296] Setting OutFile to fd 1 ...
I0516 14:24:13.513240   28241 out.go:348] isatty.IsTerminal(1) = true
I0516 14:24:13.513242   28241 out.go:309] Setting ErrFile to fd 2...
I0516 14:24:13.513244   28241 out.go:348] isatty.IsTerminal(2) = true
I0516 14:24:13.513294   28241 root.go:322] Updating PATH: /Users/dineshsankuri/.minikube/bin
I0516 14:24:13.513506   28241 out.go:303] Setting JSON to false
I0516 14:24:13.535424   28241 start.go:115] hostinfo: {"hostname":"Emplifis-MacBook-Pro.local","uptime":15888,"bootTime":1652709565,"procs":468,"os":"darwin","platform":"darwin","platformFamily":"Standalone Workstation","platformVersion":"12.0.1","kernelVersion":"21.1.0","kernelArch":"arm64","virtualizationSystem":"","virtualizationRole":"","hostId":"283ed594-67ba-5a7a-afbd-6537a2c07502"}
W0516 14:24:13.535530   28241 start.go:123] gopshost.Virtualization returned error: not implemented yet
I0516 14:24:13.556486   28241 out.go:177] 😄  minikube v1.26.0-beta.0 on Darwin 12.0.1 (arm64)
I0516 14:24:13.595928   28241 notify.go:193] Checking for updates...
I0516 14:24:13.596059   28241 driver.go:358] Setting default libvirt URI to qemu:///system
I0516 14:24:13.616017   28241 out.go:177] ✨  Using the qemu2 (experimental) driver based on user configuration
I0516 14:24:13.652781   28241 start.go:284] selected driver: qemu2
I0516 14:24:13.652787   28241 start.go:806] validating driver "qemu2" against <nil>
I0516 14:24:13.652801   28241 start.go:817] status for qemu2: {Installed:false Healthy:false Running:false NeedsImprovement:false Error:stat /opt/homebrew/Cellar/qemu/6.2.0_1/share/qemu/edk2-aarch64-code.fd: no such file or directory Reason: Fix:Install uefi firmware Doc:https://minikube.sigs.k8s.io/docs/reference/drivers/qemu2/ Version:}
I0516 14:24:13.671828   28241 out.go:177]
W0516 14:24:13.691417   28241 out.go:239] 🤷  Exiting due to PROVIDER_QEMU2_NOT_FOUND: The 'qemu2' provider was not found: stat /opt/homebrew/Cellar/qemu/6.2.0_1/share/qemu/edk2-aarch64-code.fd: no such file or directory
W0516 14:24:13.691501   28241 out.go:239] 💡  Suggestion: Install uefi firmware
W0516 14:24:13.691528   28241 out.go:239] 📘  Documentation: https://minikube.sigs.k8s.io/docs/reference/drivers/qemu2/
I0516 14:24:13.728811   28241 out.go:177]
```

### Operating System

macOS (Default)

### Driver

N/A

### PR DESCRIPTION
**Pull request by sharifelgamal**
This PR figures out the location of the qemu firmware path dynamically, rather than hardcoding a version path. Note that this still requires qemu to be installed by homebrew. 

This also allows x86_64 macOS machines to take advantage of hvf hardware acceleration for the qemu2 driver.

Fixes #14176

### CODE REVIEW

**Review by klaases — APPROVED** [review]
LGTM

File: pkg/minikube/registry/drvs/qemu2/qemu2.go [hunk 1]
```diff
@@ -64,17 +66,30 @@ func qemuSystemProgram() (string, error) {
 
 func qemuFirmwarePath() (string, error) {
 	arch := runtime.GOARCH
+	// For macOS, find the correct brew installation path for qemu firmware
+	if runtime.GOOS == "darwin" {
+		p := "/usr/local/Cellar/qemu"
```
Comment (afbjorklund) [inline pkg/minikube/registry/drvs/qemu2/qemu2.go:71]: I would probably used a switch here and added "amd64" here (like in the Linux version), but ok

**Review by afbjorklund — APPROVED** [review]
> `	// kvm acceleration doesn't currently work for linux, it's incompatible with our chosen CPU`

Weird about the KVM, I think it was supposed to use "host" as the cpu

Also had to set "gic-version" (for the real ARM host), whatever that is.

```shell
if [ -e /dev/kvm -a "`arch`" = "aarch64" ]; then
  options="-enable-kvm -cpu host -machine gic-version=3"
else
  options="-cpu cortex-a53"
fi
```

File: pkg/minikube/registry/drvs/qemu2/qemu2.go (hunk 1 above)
Comment (sharifelgamal) [reply pkg/minikube/registry/drvs/qemu2/qemu2.go]: good call, switch added

**Comment by sharifelgamal** [comment]
> > `	// kvm acceleration doesn't currently work for linux, it's incompatible with our chosen CPU`
> 
> Weird about the KVM, I think it was supposed to use "host" as the cpu
> 
> Also had to set "gic-version" (for the real ARM host), whatever that is.
> 
> ```shell
> if [ -e /dev/kvm -a "`arch`" = "x86_64" ]; then
>   options="-enable-kvm"
> else
>   options=""
> fi
> 
> if [ -e /dev/kvm -a "`arch`" = "aarch64" ]; then
>   options="-enable-kvm -cpu host -machine gic-version=3"
> else
>   options="-cpu cortex-a53"
> fi
> ```
> 
> Using `-accel kvm` instead makes it more similar to the other platforms.

Adding `gic-version=3` was the trick. I had tried making the CPU host, but that had thrown all sorts of errors i wasn't prepared to debug. kvm acceleration works now.

File: pkg/drivers/qemu/qemu.go [hunk 2]
```diff
@@ -273,19 +272,19 @@ func parsePortRange(rawPortRange string) (int, int, error) {
 
 	minPort, err := strconv.Atoi(portRange[0])
 	if err != nil {
-		return 0, 0, fmt.Errorf("invalid port range")
+		return 0, 0, errors.Wrap(err, "Invalid port range")
 	}
 	maxPort, err := strconv.Atoi(portRange[1])
```
Comment (spowelljr) [inline pkg/drivers/qemu/qemu.go:277]: We don't have any check verifying that `portRange` has a length of two, it's probably a small chance but would probably be worth checking the length and failing with `malformed port range`.

**Review by spowelljr — CHANGES_REQUESTED** [review]

File: pkg/drivers/qemu/qemu.go (hunk 2 above)
Comment (sharifelgamal) [reply pkg/drivers/qemu/qemu.go:281]: Good call, fixed.

**Review by spowelljr — APPROVED** [review]

**Pull request merged**